### PR TITLE
Fix: eigenvalues returns column vector

### PR DESCRIPTION
**Pull request by WardBrian**
## Summary

Eigen's eigensolvers return column vectors, but the current return type is set to a row vector. This causes issues such as https://github.com/stan-dev/stan/issues/3199

## Release notes

Fixed an issue with the return type of `eigenvalues` being incorrect

## Checklist

- [x] Math issue: Closes stan-dev/stan#3199

- [x] Copyright holder: Simons Foundation

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [x] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - dependencies checks pass, (`make test-math-dependencies`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Coding-Style-and-Idioms) checks (`make cpplint`)

- [x] the code is written in idiomatic C++ and changes are documented in the doxygen

- [x] the new changes are tested

### CODE REVIEW

File: stan/math/prim/fun/eigenvalues.hpp [hunk 1]
```diff
@@ -17,7 +17,7 @@ namespace math {
  */
 template <typename EigMat, require_eigen_matrix_dynamic_t<EigMat>* = nullptr,
           require_not_vt_complex<EigMat>* = nullptr>
-inline Eigen::Matrix<complex_return_t<value_type_t<EigMat>>, 1, -1> eigenvalues(
+inline Eigen::Matrix<complex_return_t<value_type_t<EigMat>>, -1, 1> eigenvalues(
```
Comment (syclik) [inline stan/math/prim/fun/eigenvalues.hpp:20]: The change looks good, but before merging, I'd like to look at the types and tests for the function to see how this wasn't trapped prior to this merge.

**Review by syclik — CHANGES_REQUESTED** [review]
"Request changes" turned on so I have a little bit of time to look at how the return types could have gotten through.

**Comment by WardBrian** [comment]
@syclik - anything I can do to help?

**Comment by syclik** [comment]
Thanks! I'm approving and let's get this merged. 

I don't think this PR needs to wait any longer. I still want to see how this is happening. 

After this issue, I'm even more adamant on not using `auto` for return types except where there's a really good reason for it. Types really provide a lot of information for the developer.

**Review by syclik — APPROVED** [review]
LGTM! Thanks!!

**Pull request merged**